internal/engine.go

```diff
@@ -51,7 +73,35 @@ func (e *Engine) initDefaultRules() {
 		&DeferRule{},
 		&MissingModPackageRule{},
 	}
-	e.registerDefaultRules()
+	e.allRules = append(e.allRules, allRules...)
+}
+
+func (e *Engine) registerDefaultRules() {
+	e.defaultRules = []LintRule{
+		NewGolangciLintRule(),
+		NewDeprecateFuncRule(),
+		NewEarlyReturnOpportunityRule(),
+		NewSimplifySliceExprRule(),
+		NewUnnecessaryConversionRule(),
+		NewLoopAllocationRule(),
+		NewEmitFormatRule(),
+		NewDetectCycleRule(),
+		NewGnoSpecificRule(),
+		NewRepeatedRegexCompilationRule(),
+		NewUselessBreakRule(),
+		NewDeferRule(),
+		NewMissingModPackageRule(),
+	}
+	e.rules = append(e.rules, e.defaultRules...)
+}
+
+func (e *Engine) findRule(name string) *LintRule {
+	for _, rule := range e.rules {
+		if rule.Name() == name {
+			return &rule
+		}
+	}
+	return nil
```

Comment:
Returning the address of a loop variable may lead to unexpected behavior. Please remove the pointer here.
```suggestion
func (e *Engine) findRule(name string) LintRule {
    for _, rule := range e.rules {
        if rule.Name() == name {
            return rule
        }
    }
    return nil
```

Alternatively, you could use a map to reduce the complexity from O(n) to O(1)

Reply:
went through maps only with this commit https://github.com/gnoverse/tlin/pull/91/commits/ca6a4e47b441e7a4a9107b3b678c179f195956c7, tell me what you think about it